trie/utils.go

```diff
@@ -31,169 +33,93 @@ package trie
 //
 // Besides, it's also used for recording the original value of the nodes
 // when they are resolved from the disk. The pre-value of the nodes will
-// be used to construct reverse-diffs in the future.
+// be used to construct trie history in the future.
 //
 // Note tracer is not thread-safe, callers should be responsible for handling
 // the concurrency issues by themselves.
 type tracer struct {
```

Comment:
nitpick, if this file only contains the `tracer`, can we rename it to `tracer.go`? It's a fairly bad practice to name things `util`, `data` and similar because they are too generic to be of use when searching for things.